### DISCUSSION TITLE
Secure WebSockets (wss)

### DISCUSSION THREAD

**Discussion by claytonEtillman** [fonsp/Pluto.jl#1143] [Questions about using Pluto] [answered]
Hi Fons et alii,

I am trying to embed a Pluto notebook inside a web snippet using the CMS framework of Odoo (previously OpenERP). It sits behind an nginx reverse proxy. I was able to figure out what I needed to have nginx put in the headers to upgrade to WebSocket, but I think there may be an implementation issue with WebSocket in the Pluto code. That or I'm overlooking something on my end, of course, but I've checked about every other angle so here I am.

I'm trying to connect a MIDI controller through Web MIDI API. I am able to get a success on the promise navigator.requestMIDIAccess(), but then when I try to access the MIDIAccess.inputs I can't see anything. The Web recommendation is to only allow MIDI access through https, which I have been able to configure this far.

I saw the request from Pluto was in ws protocol. I think it needs to be in wss for my use case. I skimmed over the Pluto.jl code last night, but couldn't find where to make that switch. Any assistance would be appreciated.

Thanks,
Clayton

**Comment by fonsp** [comment, marked as answer]
I don't understand your question, perhaps you can try rephrasing it in a way that we can read it without having to learn what the MIDI API is.

About `ws`/`wss`, it should match the `http`/`https` protocol being used. This is done automatically, code is here https://github.com/fonsp/Pluto.jl/blob/099f40a402de7215e1092ea740237082755a9d8a/frontend/common/PlutoConnection.js#L182-L186

**Reply by claytonEtillman** [reply]
Thanks for the reply. That's basically what I was looking for. It looks like the problem is likely somewhere else.  I'll look into my nginx configuration.

The issue is basically I can get the code here to work: https://codepen.io/georgemandis/pen/XqYNNN

, but when I try to run the same code in Pluto from my website, my MIDI keyboard doesn't show in the list of available instruments.

I also wanted to say thanks for developing Pluto. It's pretty awesome! I'm very excited to be able to use Julia dynamically in the browser.
